sklearnex/ensemble/_forest.py

```diff
@@ -58,38 +56,58 @@
 from onedal.ensemble import RandomForestClassifier as onedal_RandomForestClassifier
 from onedal.ensemble import RandomForestRegressor as onedal_RandomForestRegressor
 from onedal.primitives import get_tree_state_cls, get_tree_state_reg
-from onedal.utils.validation import _num_features, _num_samples
-from sklearnex._utils import register_hyperparameters
+from onedal.utils.validation import _num_features
 
 from .._config import get_config
 from .._device_offload import dispatch, wrap_output_data
-from .._utils import PatchingConditionsChain
+from .._utils import PatchingConditionsChain, register_hyperparameters
 from ..base import oneDALEstimator
-from ..utils._array_api import get_namespace
-from ..utils.validation import check_n_features, validate_data
+from ..utils._array_api import enable_array_api, get_namespace
+from ..utils.class_weight import _compute_class_weight
+from ..utils.validation import _check_sample_weight, assert_all_finite, validate_data
 
 if sklearn_check_version("1.2"):
     from sklearn.utils._param_validation import Interval
-if sklearn_check_version("1.4"):
-    from daal4py.sklearn.utils import _assert_all_finite
+
+
+if sklearn_check_version("1.6"):
+    _check_array = partial(
+        check_array,
+        ensure_all_finite=False,
+        dtype=None,
+        ensure_2d=False,
+        ensure_min_samples=0,
+        ensure_min_features=0,
+        accept_sparse=True,
+    )
+else:
+    _check_array = partial(
+        check_array,
+        force_all_finite=False,
+        dtype=None,
+        ensure_2d=False,
+        ensure_min_samples=0,
+        ensure_min_features=0,
+        accept_sparse=True,
+    )
```

Comment:
```suggestion
__check_kwargs = {
  "dtype": None,
  "ensure_2d": False,
  "ensure_min_samples": 0,
  "ensure_min_features": 0,
  "accepts_sparse": True,
}
if sklearn_check_version("1.6"):
  __check_kwargs["ensure_all_finite"] = False
else:
  __check_kwargs["force_all_finite"] = False
_check_array = partial(check_array, **__check_kwargs)
```

Or any other pattern that avoids the duplication...